### PR TITLE
feat: july milestone

### PR DESCRIPTION
**Pull request by dhmlau**
Trying something different. Opening this PR early so that we can get more inputs from @strongloop/loopback-maintainers and the community. The general theme is docs issues takedown.  Please edit this branch directly or comment on the PR.

Reference: [Q3 roadmap](https://github.com/strongloop/loopback-next/blob/master/docs/ROADMAP.md#q3-2020-roadmap)

## Checklist

👉 [Read and sign the CLA (Contributor License Agreement)](https://cla.strongloop.com/agreements/strongloop/loopback-next) 👈

- [ ] `npm test` passes on your machine
- [ ] New tests added or existing tests modified to cover all changes
- [ ] Code conforms with the [style guide](http://loopback.io/doc/en/contrib/style-guide.html)
- [ ] API Documentation in code was updated
- [ ] Documentation in [/docs/site](../tree/master/docs/site) was updated
- [ ] Affected artifact templates in `packages/cli` were updated
- [ ] Affected example projects in `examples/*` were updated

👉 [Check out how to submit a PR](https://loopback.io/doc/en/lb4/submitting_a_pr.html) 👈

### CODE REVIEW

File: docs/_jul_milestone_.md [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+## Overview
+
+- Docs issue takedown!
+
+## Goals
+
+- [ ] Docs improvements
+
+  _Please add your suggestions here. Also see
+  https://github.com/strongloop/loopback-next/issues?q=is%3Aissue+is%3Aopen+label%3ADocs+label%3A2020Q3_
```
Comment (bajtos) [inline docs/_jul_milestone_.md:12]: - Hide `@loopback/openapi-v3` from docs, examples and CLI templates #5692

File: docs/_jul_milestone_.md [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+- [ ] Feature parity
+
+  - [ ] Execute raw NoSQL queries (e.g. MongoDB) #3342
+
+- [ ] Fix GVT issues
+  - [ ] [GSSC-GVT]English String when input command apic lb4 app -h #5654
+  - [ ] [GSSC-GVT]English String when input command apic lb4 repository #5656
+  - [ ] [GSSC-GVT]English String when input command apic lb4 model #5657
+
```
Comment (bajtos) [inline docs/_jul_milestone_.md:22]: Chores:

- update dependency mem-fs-editor to v7 #5624

Comment (bajtos) [reply docs/_jul_milestone_.md:22]: Tooling improvements:

- [api-extractor] Incorrect handling of interfaces enhancing a class with the same name https://github.com/microsoft/rushstack/issues/1921

**Comment by hacksparrow** [comment]
I vote for https://github.com/strongloop/loopback-next/issues/5697. It is mostly a refactoring task, but will have a drastic effect on the state of our documentation.

Most of our users are looking for a REST framework documentation, but what we have is a documentation about the general-purpose API framework with the REST features documented in parts in different sections of the grand document like a jigsaw puzzle.

With a REST extension, we would have created the entity whose documentation our users are looking for. It also makes it clear LoopBack is a general-purpose API framework that can be used to build  REST/OpenAPI, GraphQL and other APIs.

I would like to take up this task myself.

**Comment by hacksparrow** [comment]
Let's dedicate some time to analyze and discuss this PR - https://github.com/strongloop/loopback-next/pull/5366.

File: docs/_jul_milestone_.md [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+- Docs issue takedown!
+
+## Goals
+
+- [ ] Docs improvements
+
+  _Please add your suggestions here. Also see
+  https://github.com/strongloop/loopback-next/issues?q=is%3Aissue+is%3Aopen+label%3ADocs+label%3A2020Q3_
+
```
Comment (agnes512) [inline docs/_jul_milestone_.md:13]: Add missing LoopBack 4 Types docs https://github.com/strongloop/loopback-next/issues/5063
Docs: how to configure a custom table/collection name different from model class name #2197 ( I think I've done this for those main connectors but it'd be better to check it thoroughly)
Maybe a good stretch goal: couchDB connector docs #5267

Comment (dhmlau) [reply docs/_jul_milestone_.md:13]: Regression: MongoError: unknown operator: $$eq  https://github.com/strongloop/loopback-connector-mongodb/issues/532

**Comment by dhmlau** [comment]
> I vote for #5697. It is mostly a refactoring task, but will have a drastic effect on the state of our documentation.
> 
> Most of our users are looking for a REST framework documentation, but what we have is a documentation about the general-purpose API framework with the REST features documented in parts in different sections of the grand document like a jigsaw puzzle.
>
> With a REST extension, we would have created the entity whose documentation our users are looking for. It also makes it clear LoopBack is a general-purpose API framework that can be used to build REST/OpenAPI, GraphQL and other APIs.

@hacksparrow, I'm a bit reserved about getting #5697 on the July milestone, probably because I (as a user) cannot see the benefits of having a separate REST extension. I'm thinking if the items we've listed in the [Q3 roadmap](https://github.com/strongloop/loopback-next/blob/master/docs/ROADMAP.md#q3-2020-roadmap) should have a higher priority over #5697 for the Q3 time frame.
Would also like to hear from @raymondfeng and @bajtos.

**Comment by bajtos** [comment]
I agree with @dhmlau regarding #5697. The scope of the issue looks big to me, I am not convinced the benefits to our users will justify the effort required to make the changes happen. We have bigger fish to fry. 

> Most of our users are looking for a REST framework documentation, but what we have is a documentation about the general-purpose API framework with the REST features documented in parts in different sections of the grand document like a jigsaw puzzle.

Let's find a way how to improve our documentation to make it easier for our users to find information about REST features, so that it does not feel like a jigsaw puzzle anymore.

File: docs/_jul_milestone_.md [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+## Goals
+
+- [ ] Docs improvements #5113
+
+  - [ ] Hide @loopback/openapi-v3 from docs, examples and CLI templates #5692
+  - [ ] Docs: how to configure a custom table/collection name different from
+        model class name #2197
+  - [ ] Improve and refactor the authentication documentations #5294
+
```
Comment (bajtos) [inline docs/_jul_milestone_.md:18]: Can we add the following task?

- [Docs] Add "How the documentation is organized" section #5752

File: docs/_jul_milestone_.md [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+- [ ] Feature parity
+
+  - [ ] Execute raw NoSQL queries (e.g. MongoDB) #3342
+
+- [ ] Chores/Tooling improvements
+
+  - [ ] update dependency mem-fs-editor to v7 #5624
+  - [ ] [api-extractor] Incorrect handling of interfaces enhancing a class with
+        the same name microsoft/rushstack#1921
```
Comment (bajtos) [inline docs/_jul_milestone_.md:60]: We found a (workaround) solution that does not involve interfaces enhancing a class. Let's remove this task from the milestone.

File: docs/_jul_milestone_.md [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+## Goals
+
+- [ ] Docs improvements #5113
+
+  - [ ] Hide @loopback/openapi-v3 from docs, examples and CLI templates #5692
+  - [ ] Docs: how to configure a custom table/collection name different from
+        model class name #2197
+  - [ ] Improve and refactor the authentication documentations #5294
+  - [ ] [Docs] Add "How the documentation is organized" section #5752
+
```
Comment (dhmlau) [inline docs/_jul_milestone_.md:19]: Probably need to defer `Clean up extension template and docs` #5336 to Jul.

**Comment by dhmlau** [comment]
@agnes512, since you'll be the scrum master for next month, could you please open a new issue for the milestone? Thanks.

**Comment by agnes512** [comment]
Closing this PR as #5837 is opened.

**Pull request closed without merging**